### PR TITLE
RtpsRelay: fixed static, asan, Windows issues

### PR DESCRIPTION
**Pull request by mitza-oci**

### CODE REVIEW

File: dds/DCPS/RTPS/Spdp.cpp [hunk 1]
```diff
@@ -1705,15 +1705,15 @@ Spdp::SpdpTransport::handle_input(ACE_HANDLE h)
...
   } else if (bytes == 0) {
-    return -1;
+    return 0;
   } else {
     ACE_DEBUG((
-          LM_ERROR,
-          ACE_TEXT("(%P|%t) ERROR: Spdp::SpdpTransport::handle_input() - ")
+          LM_WARNING,
+          ACE_TEXT("(%P|%t) WARNING: Spdp::SpdpTransport::handle_input() - ")
           ACE_TEXT("error reading from %C socket %p\n")
```
Comment (simpsont-oci) [inline dds/DCPS/RTPS/Spdp.cpp]: Was this impacting tests, or some other issue? Do we need to remove the word error from the debug message as well?

Comment (jrw972) [reply dds/DCPS/RTPS/Spdp.cpp]: It impacts the new tests for RtpsRelay.  We observed "connection reset" errors.  Keeping error in the text is appropriate.

File: tools/rtpsrelay/RelayHandler.cpp [hunk 2]
```diff
@@ -76,7 +76,9 @@ int RelayHandler::handle_input(ACE_HANDLE)
 
   bytes_received_ += bytes;
 
-  const auto data_block = new ACE_Data_Block(bytes, ACE_Message_Block::MB_DATA, static_cast<const char*>(iov.iov_base), 0, 0, 0, 0);
+  const auto data_block =
```
Comment (jrw972) [inline tools/rtpsrelay/RelayHandler.cpp]:
```suggestion
  const auto data_block =
```

File: dds/DCPS/RTPS/Spdp.cpp (hunk 1 above)
Comment (mitza-oci) [reply dds/DCPS/RTPS/Spdp.cpp]: https://github.com/DOCGroup/autobuild/blob/891e9c5fe2bab56ca54a4ecd8f267f574aca55c0/common/prettify.pm#L1468

Comment (simpsont-oci) [reply dds/DCPS/RTPS/Spdp.cpp]: Gotcha. Always wondered about the full list. Now I know not to make any debug messages about 1970/80's cartoon show "fatalbert" or the value of my eclectically-named constant "INISFAIL:"

**Pull request merged**